### PR TITLE
ocamlPackages.memtrace: init at 0.2.2

### PR DESCRIPTION
**Pull request by Niols**
###### Description of changes

Streaming client for OCaml's Memprof

https://github.com/janestreet/memtrace/blob/v0.2.2/CHANGES.md

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/memtrace/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "janestreet";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-y/Xz04CMFfRIzrDzGot16zEQsBMNc4J5s/q0VERcj04=";
+  };
+
+  duneVersion = "3";
```
[the remark this answers is not in the document]
Comment (Niols) [reply pkgs/development/ocaml-modules/memtrace/default.nix]: Out of curiosity: why remove `duneVersion`?

[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/ocaml-modules/memtrace/default.nix]: No information means: “the default version of dune is OK“. Explicit version information may means: “this specific version is needed”. This has an impact when the default version changes.

**Comment by anmonteiro** [comment]
Why didn't you track the latest version instead? https://github.com/janestreet/memtrace/releases/tag/v0.2.3

**Comment by Niols** [comment]
Because I stupidly followed the last release, assuming tags would follow. I will provide an update.

**Comment by Niols** [comment]
cf https://github.com/NixOS/nixpkgs/pull/238267

**Pull request merged**